### PR TITLE
[CPPCHECK] fix various warnings

### PR DESCRIPTION
**Pull request by lrusak**
This PR corrects some simple issues found by cppcheck that runs in our CI.

The point of this PR is not only to fix these issues, but to encourage others to actively check the reports for new warnings and to fix old ones.

For those unaware the cppcheck analysis can be found here: https://jenkins.kodi.tv/view/Reports/job/LINUX-64-GL-Static-Analysis/lastBuild/cppcheck/

The following issues are being addressed:
 1) remove necessary c_str() usage
 2) fix unique_ptr usage after std::move
 3) fix non bool return in a bool function
 4) add constructors where needed

### CODE REVIEW

File: xbmc/dbwrappers/DatabaseQuery.cpp [hunk 1]
```diff
@@ -246,7 +246,7 @@ std::string CDatabaseQueryRule::FormatParameter(const std::string &operatorStrin
     parameter = " IN (" + parameter + ")";
   }
   else
-    parameter = db.PrepareSQL(operatorString.c_str(), ValidateParameter(param).c_str());
+    parameter = db.PrepareSQL(operatorString, ValidateParameter(param));
```
Comment (wsnipex) [inline xbmc/dbwrappers/DatabaseQuery.cpp:249]: this is a C wrapper, so likely needs c_str

File: xbmc/dbwrappers/DatabaseQuery.cpp [hunk 2]
```diff
@@ -256,7 +256,7 @@ std::string CDatabaseQueryRule::FormatParameter(const std::string &operatorStrin
       CDateTimeSpan span;
       span.SetFromPeriod(param);
       date-=span;
-      parameter = db.PrepareSQL(operatorString.c_str(), date.GetAsDBDate().c_str());
+      parameter = db.PrepareSQL(operatorString, date.GetAsDBDate());
```
Comment (wsnipex) [inline xbmc/dbwrappers/DatabaseQuery.cpp:259]: same

File: xbmc/dbwrappers/DatabaseQuery.cpp (hunk 1 above)
Comment (lrusak) [reply xbmc/dbwrappers/DatabaseQuery.cpp:249]: https://github.com/xbmc/xbmc/blob/a644f0674c06cb6080612ad9c9d8b0d5a9f8a41a/xbmc/dbwrappers/Database.h#L114

Comment (lrusak) [reply xbmc/dbwrappers/DatabaseQuery.cpp:249]: Ah, va_args, meh, the first one can still be removed though.

File: xbmc/playlists/SmartPlayList.cpp [hunk 3]
```diff
@@ -831,7 +831,7 @@ std::string CSmartPlaylistRule::FormatParameter(const std::string &operatorStrin
   if (m_field == FieldTime || m_field == FieldAlbumDuration)
   { // translate time to seconds
     std::string seconds = std::to_string(StringUtils::TimeStringToSeconds(param));
-    return db.PrepareSQL(operatorString.c_str(), seconds.c_str());
+    return db.PrepareSQL(operatorString, seconds);
```
Comment (wsnipex) [inline xbmc/playlists/SmartPlayList.cpp:834]: same var_args issue here

**Review by garbear — APPROVED** [review]
Everything looks good assuming you got the vargs part worked out

**Review by wsnipex — APPROVED** [review]

**Pull request merged**